app/src/main/java/org/mozilla/fenix/components/FindInPageIntegration.kt

```diff
@@ -5,38 +5,111 @@
         store.state.findCustomTabOrSelectedTab(sessionId)?.let { tab ->
-            // Always hide the toolbar and display find in page query
-            toolbar.visibility = View.GONE
+            prepareLayoutForFindBar()
+
             view.visibility = View.VISIBLE
             (feature as FindInPageFeature).bind(tab)
-            view.layoutParams.height = toolbar.height
+            view.layoutParams.height = toolbarInfo.toolbar.height
         }
     }
+
+    @VisibleForTesting
+    internal fun restorePreviousLayout() {
+        toolbarInfo.toolbar.isVisible = true
+
+        val engineViewParent = getEngineViewParent()
+        val engineViewParentParams = getEngineViewsParentLayoutParams()
+        if (toolbarInfo.isToolbarPlaceAtTop) {
+            if (toolbarInfo.isToolbarDynamic) {
+                engineViewParent.translationY = toolbarInfo.toolbar.height.toFloat()
+                engineViewParentParams.bottomMargin = 0
+            } else {
+                engineViewParent.translationY = 0f
+            }
+        } else {
+            if (toolbarInfo.isToolbarDynamic) {
+                engineViewParentParams.bottomMargin = 0
+            }
+        }
+    }
+
+    @VisibleForTesting
+    internal fun prepareLayoutForFindBar() {
+        toolbarInfo.toolbar.isVisible = false
+
+        val engineViewParent = getEngineViewParent()
+        val engineViewParentParams = getEngineViewsParentLayoutParams()
+        if (toolbarInfo.isToolbarPlaceAtTop) {
+            if (toolbarInfo.isToolbarDynamic) {
+                // With a dynamic toolbar the EngineView extends to the entire (top and bottom) of the screen.
+                // And now with the toolbar expanded it is translated down immediately below the toolbar.
+                engineViewParent.translationY = 0f
+                engineViewParentParams.bottomMargin = toolbarInfo.toolbar.height
+            } else {
+                // With a fixed toolbar the EngineView is anchored below the toolbar with 0 Y translation.
+                engineViewParent.translationY = -toolbarInfo.toolbar.height.toFloat()
+            }
+        } else {
+            // With a bottom toolbar the EngineView is already anchored to the top of the screen.
+            // Need just to ensure space for the find in page bar under the engineView.
+            engineViewParentParams.bottomMargin = toolbarInfo.toolbar.height
+        }
+    }
+
+    @VisibleForTesting
+    internal fun getEngineViewParent() = engineView.asView().parent as View
+
+    @VisibleForTesting
+    internal fun getEngineViewsParentLayoutParams() = getEngineViewParent().layoutParams as MarginLayoutParams
+
+    /**
+     * Holder of all details needed about the Toolbar.
+     * Used to modify the layout of BrowserToolbar while the find in page bar is shown.
+     */
+    data class ToolbarInfo(
+        val toolbar: BrowserToolbar,
+        val isToolbarDynamic: Boolean,
+        val isToolbarPlaceAtTop: Boolean
```

Comment:
nit: did you meant isToolbarPlace**d**AtTop ?

Reply:
Indeed. Thank you!